ide/che-core-ide-stacks/src/main/resources/stacks.json

```diff
@@ -1,4 +1,138 @@
 [
+  {
+    "id": "java-theia-docker",
+    "creator": "ide",
+    "name": "Java Theia (docker)",
+    "description": "Java + Theia in a sidecar container (docker)",
+    "scope": "general",
+    "tags": [
+      "Java 1.8, Theia"
+    ],
+    "workspaceConfig": {
+      "environments": {
+        "default": {
+          "machines": {
+            "theia": {
+              "attributes": {},
+
+              "servers": {
+                "theia": {
+                  "protocol": "http",
+                  "port": "3000",
+                  "attributes": {
+                    "type": "ide"
+                  }
+                }
+              },
+              "volumes": {
+                "projects": {
+                  "path": "/projects"
+                }
+              },
+              "installers": [],
+              "env": {}
+            },
+            "dev-machine": {
+              "attributes": {},
+              "servers": {
+                "theia-dev": {
+                  "protocol": "http",
+                  "port": "3030"
+                }
+              },
+              "volumes": {
+                "projects": {
+                  "path": "/projects"
+                }
+              },
+              "installers": [],
+              "env": {}
+            }
+          },
+          "recipe": {
+            "type": "compose",
+            "content": "services:\n theia:\n  image: theiaide/theia\n  entrypoint:\n  - yarn\n  - theia\n  - start\n  - /projects\n  - --hostname=0.0.0.0\n  mem_limit: 1073741824\n dev-machine:\n  image: eclipse/che-dev:nightly\n  mem_limit: 2147483648\n  depends_on:\n    - theia",
+            "contentType": "application/x-yaml"
+          }
+        }
+      },
+      "defaultEnv": "default",
+      "name": "theia",
+      "projects": [],
+      "commands": []
+    },
+    "stackIcon": {
+      "name": "type-java.svg",
```

Comment:
should we replace the logo by something more theia ?

Reply:
I've asked but they have nothing to share.

Reply:
maybe we could do like for C++ with a generic "text" logo waiting the official logo in svg format.
Because in the che dev machine we have other tools than java.

Reply:
I can put this 
https://github.com/eclipse/che/blob/master/dashboard/src/assets/svg/type-blank.svg
or this 
https://github.com/eclipse/che/blob/master/ide/che-core-ide-stacks/src/main/resources/stacks-images/type-che.svg
what do you prefer?

Reply:
https://gist.github.com/benoitf/aca50dd40c1a2a25c20f37ed63b9571c

Reply:
done